### ISSUE TITLE
 loss_ = -1 * np.power(1 - pro_, self._gamma) * np.log(pro_) is wrong

### ISSUE THREAD

**Issue by Sunting78** [unsky/focal-loss#14]
loss_ = -1 * np.power(1 - pro_, self._gamma) * np.log(pro_) is wrong . The pro_ should be pt, right?

**Comment by unsky** [comment]
i do not understand your mean, the codes in metric?
the loss metric just a display and dont affact the model.

if your loss_ is  in the metric.py
pro_ and pt all will be ok.
pro_ is to metric the total examples loss,
but pt is to metric the positive examples loss.

**Comment by Sunting78** [comment]
Yes,Iknow. In focal loss.py  loss_ = -1 * np.power(1 - pro_, self.gamma) * np.log(pro) .I know the loss metric just a display and dont affact the model. But I think the softmaxloss = -log(pt) pt is the positive examples softmax pro.

**Comment by unsky** [comment]
no. in standard softmax with ce loss is to metric total examples loss.

**Comment by Sunting78** [comment]
I think softmaxloss = -sum(yilog(pi)). But label is one hot.  So  softmaxloss = -log(pt)

**Comment by unsky** [comment]
YES. -1*y*log(pro_)=-log(pt), standard softmax with ce loss is to metric positive  loss.
 but in general,i like display the total loss and positive loss,these loss can help us to adjust the params setting. you are right in one-hot label.
